servicebroker/test/models/binding_test.js

```diff
@@ -0,0 +1,183 @@
+'use strict';
+var expect = require('chai').expect;
+var serviceInstance = require('../../lib/models')().service_instance;
+var binding = require('../../lib/models')().binding;
+
+var fake_serviceInstanceId = "test_serviceinstance";
+var fake_orgId = "test_org";
+var fake_spaceId = "test_space";
+var fake_appId = "test_app";
+var fake_bindingId = "test_binding";
+
+var fake_appId2 = "test_app2";
+var fake_bindingId2 = "test_binding2";
+
+var fake_service_condition = {
+  'serviceInstanceId': fake_serviceInstanceId,
+  'orgId': fake_orgId,
+  'spaceId': fake_spaceId,
+  where: { 'serviceInstanceId': fake_serviceInstanceId, 'orgId': fake_orgId, 'spaceId': fake_spaceId },
+
+};
+var fake_binding_condition = {
+  'serviceInstanceId': fake_serviceInstanceId,
+  'appId': fake_appId,
+  'bindingId': fake_bindingId,
+  'timestamp': new Date().getTime() 
+};
+var where_condition = {where: { 'serviceInstanceId': fake_serviceInstanceId, 'appId': fake_appId, 'bindingId': fake_bindingId }};
+var fake_binding_condition2 = {
+  'serviceInstanceId': fake_serviceInstanceId,
+  'appId': fake_appId2,
+  'bindingId': fake_bindingId,
+  'timestamp': new Date().getTime() 
+};
+
+describe('Binding Model Definition Test Suite', function() {
+
+  before(function(done) {
+    serviceInstance.sequelize.sync().then(function(result) {
+      binding.sequelize.sync().then(function(result) {
+        done();
+      }).catch(function(error1) {
+        console.log("Failed to sync model binding, error: " + error1);
+        done(error1);
+      });
+    }).catch(function(error) {
+      console.log("Failed to sync model serviceInstance, error: " + error);
+      done(error);
+    });
+  });
+
+  beforeEach(function(done) {
+    return binding.truncate({ cascade: true }).then(function(result) {
```

Comment:
With cascade: true, you should be able to truncate the service_instance table and have it wipe the binding table.
We should not be doing a findOrCreate here but in the contexts that require an item in the DB.